activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/MessageDatabase.java

```diff
@@ -1718,7 +1722,10 @@ private void checkpointUpdate(final boolean cleanup) throws IOException {
 
         metadata.state = OPEN_STATE;
         metadata.producerSequenceIdTrackerLocation = checkpointProducerAudit();
-        metadata.ackMessageFileMapLocation = checkpointAckMessageFileMap();
+        if (metadata.ackMessageFileMapDirtyFlag.get() || (metadata.ackMessageFileMapLocation == null)) {
```

Comment:
isn't the `ackMessageFileMapDirtyFlag` redundant to the `ackMessageFileMapMod` local variable down below?